autogen/agentchat/contrib/vectordb/base.py

```diff
@@ -0,0 +1,229 @@
+            upsert: bool | Whether to update the document if it exists. Default is False.
+            kwargs: Dict | Additional keyword arguments.
+
+        Returns:
+            None
+        """
+        ...
+
+    def update_docs(self, docs: List[Document], collection_name: str = None, **kwargs) -> None:
+        """
+        Update documents in the collection of the vector database.
+
+        Args:
+            docs: List[Document] | A list of documents.
+            collection_name: str | The name of the collection. Default is None.
+            kwargs: Dict | Additional keyword arguments.
+
+        Returns:
+            None
+        """
+        ...
+
+    def delete_docs(self, ids: List[ItemID], collection_name: str = None, **kwargs) -> None:
+        """
+        Delete documents from the collection of the vector database.
+
+        Args:
+            ids: List[ItemID] | A list of document ids. Each id is a typed `ItemID`.
+            collection_name: str | The name of the collection. Default is None.
+            kwargs: Dict | Additional keyword arguments.
+
+        Returns:
+            None
+        """
+        ...
+
+    def retrieve_docs(
+        self,
+        queries: List[str],
+        collection_name: str = None,
+        n_results: int = 10,
+        distance_threshold: float = -1,
+        **kwargs,
+    ) -> QueryResults:
+        """
+        Retrieve documents from the collection of the vector database based on the queries.
+
+        Args:
+            queries: List[str] | A list of queries. Each query is a string.
+            collection_name: str | The name of the collection. Default is None.
+            n_results: int | The number of relevant documents to return. Default is 10.
+            distance_threshold: float | The threshold for the distance score, only distance smaller than it will be
+                returned. Don't filter with it if < 0. Default is -1.
+            kwargs: Dict | Additional keyword arguments.
+
+        Returns:
+            QueryResults | The query results. Each query result is a TypedDict `QueryResults`.
+            It should include the following fields:
+                - required: "ids", "contents"
+                - optional: "embeddings", "metadatas", "distances", etc.
+
+            queries example: ["query1", "query2"]
+            query results example: {
+                "ids": [["id1", "id2", ...], ["id3", "id4", ...]],
+                "contents": [["content1", "content2", ...], ["content3", "content4", ...]],
+                "embeddings": [["embedding1", "embedding2", ...], ["embedding3", "embedding4", ...]],
+                "metadatas": [["metadata1", "metadata2", ...], ["metadata3", "metadata4", ...]],
+                "distances": [["distance1", "distance2", ...], ["distance3", "distance4", ...]],
+            }
+
+        """
+        ...
+
+    def get_docs_by_ids(self, ids: List[ItemID], collection_name: str = None, include=None, **kwargs) -> GetResults:
```

Comment:
I feel it might be better to keep this API -- users might use this API to debug, e.g., if the retrieval behavior is not as expected (missing some item), then users can use this API to first check whether this item has been successfully inserted. At least VikingDB support this API.

Reply:
Thanks @siluhuang , indeed no harm to keep it and can be useful for debugging. Added back.